test/functorch/test_control_flow.py

```diff
@@ -2011,13 +2117,196 @@ def chain_fct_different_dim(inp):
+        #     0,
+        #     reverse=reverse_first
+        # )
+        
+        # self.assertEqual(result, expected_result)
+        
+    # TODO: provide an implementation for all compile modes and re-enable all test
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("compile_mode", ["none", "compile", "compile_dynamic_shape"])
+    # @parametrize("loop_type", ["for", "while"])
+    @parametrize("loop_type", ["for"])
+    # @parametrize("loop_type", ["while"])
+    @parametrize("reverse", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping the combination of combine_mode=pointwise and device=cpu
+    # as the current implementation of pointwise does only support CUDA device
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+            # and (params["device"] == torch.device("cpu") or torch.version.hip)
+        ),
+    )
+    def test_associative_scan_loop(
+        self, combine_mode, compile_mode, loop_type, reverse, device
+    ):
+        def loop_body(x, y):
+            cnt = torch.zeros_like(y[0, :])
+            if loop_type == 'while':
+                def cond_fn(ind, loop_val):
+                    return (loop_val < 5)[0]
+                
+                def body_fn(ind, loop_val):
+                    return ind + 1, loop_val + torch.abs(ind)
+                
+                new_ind, cnt = torch.while_loop(cond_fn=cond_fn, body_fn=body_fn, carried_inputs=(torch.zeros(1, dtype=torch.int32, device=cnt.device), cnt))
+            else:
+                for ind in range(10):
+                    cnt += torch.abs(y[ind])
+            return x * cnt
+
+        scan_fct = compile_mode_helper(associative_scan, compile_mode)
+
+        inp = torch.randn(3, 10, 1, device=device) * 2
+        
+        result = scan_fct(loop_body, inp, dim=0, reverse=reverse, combine_mode=combine_mode)
+        expected_result = _fake_associative_scan(
+            loop_body,
+            inp,
+            0,
             reverse=reverse,
         )
-        result1 = fct_cmp(inp)
-        self.assertEqual(result1, expected_result)
+        
+        self.assertEqual(result, expected_result)
+        
+    # TODO: provide an implementation for all compile modes and re-enable all test
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("compile_mode", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("reverse", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping combine_mode=pointwise 
+    # as the cond operation is classified to be non-pointwise
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
```

Comment:
Might just remove the combine_mode parametrize arg

Reply:
Done